File: HW/HW11/HW11.py
from collections import deque
class HW11(object):
    def floodFill(self, image, sr, sc, color):
        current = image[sr][sc]
        visited = []
        queue = deque()
        temp = [sr,sc]
        queue.append(temp)
        while queue:
            temp = queue.pop()
            sr = temp[0]
            sc = temp[1]
            if sr < 0 or sr >= len(image) or sc < 0 or sc >= len(image[0]) or image[sr][sc] != current:
                continue
            image[sr][sc] = color
            visited.append(temp)
            temp = [sr+1,sc]
            if temp not in visited:
                queue.append(temp)
            temp = [sr-1,sc]
            if temp not in visited:
                queue.append(temp)
            temp = [sr,sc+1]
            if temp not in visited:
                queue.append(temp)
            temp = [sr,sc-1]
            if temp not in visited:
                queue.append(temp)
        return image

File: HW/HW11/HW11.java
import java.util.*;
public class HW11 {
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        int current = image[sr][sc];
        List<List<Integer>> visited = new ArrayList<>();
        Queue<List<Integer>> queue = new LinkedList<>();
        List<Integer> temp = new ArrayList<>(Arrays.asList(sr, sc));
        queue.add(temp);
        while (!queue.isEmpty()){
            temp = queue.poll();
            sr = temp.get(0);
            sc = temp.get(1);
            if (sr < 0 || sr >= image.length || sc < 0 || sc >= image[0].length ||  image[sr][sc] != current){
            }
            else{
                image[sr][sc] = color;
                visited.add(temp);
                temp = new ArrayList<>(Arrays.asList(sr+1, sc));
                if (!visited.contains(temp)){
                    queue.add(temp);
                }
                temp = new ArrayList<>(Arrays.asList(sr-1, sc));
                if (!visited.contains(temp)){
                    queue.add(temp);
                }
                temp = new ArrayList<>(Arrays.asList(sr, sc+1));
                if (!visited.contains(temp)){
                    queue.add(temp);
                }
                temp = new ArrayList<>(Arrays.asList(sr, sc-1));
                if (!visited.contains(temp)){
                    queue.add(temp);
                }
            }
        }
        return image;
    }
}